F-LSeSim/models/kin.py

```diff
@@ -5,10 +5,36 @@
 
 
 class KernelizedInstanceNorm(nn.Module):
-
-    def __init__(self, num_features, eps=0, affine=True, device="cuda"):
+    """
+
+    Attributes:
+        num_features (int): The number of input features.
+        kernel_size (int): The size of kernel used to kernelize.
+        kernel_type (str): The kernel type. It must be either 'constant' or 'gaussian'.
+        padding (int): The padding size of cache table.
+        eps (float): A value added to the denominator for numerical stability.
+        affine (bool): A boolean value that when set to True, this module has learnable
+            affine parameters, initialized the same way as done for batch normalization.
+        device (str): The device to place the variables.
+    """
+
+    def __init__(
+        self,
+        num_features,
+        kernel_size=3,
+        kernel_type='gaussian',
+        padding=1,
+        eps=0,
+        affine=True,
+        device="cuda",
+    ):
         super(KernelizedInstanceNorm, self).__init__()
 
+        if kernel_size % 2 == 0:
```

Comment:
Additionally check `padding * 2 + 1 == kernel_size`?

Reply:
Done. If `padding` is not equal to `(kernel_size - 1) // 2`, a `ValueError` will be raised.